modules/core/shared/src/main/scala/gem/config/DynamicConfig.scala

```diff
@@ -82,6 +82,12 @@ object DynamicConfig {
 
     type GmosNorthDefinition = GmosDefinition[GmosNorthDisperser, GmosNorthFilter, GmosNorthFpu]
     type GmosSouthDefinition = GmosDefinition[GmosSouthDisperser, GmosSouthFilter, GmosSouthFpu]
+
+    final case class Gnirs(
+      disperser: Option[GnirsDisperser],
+      filter:    GnirsDisperser,
+      fpu:       Option[GnirsFpuSlit]
+    )
```

Comment:
This is for Smart Gcal correct?  Does the same case class serve both for searching and for defining entries (as for F2)?  If so you'll need to extend `SmartGcalSearchKey with SmartGcalDefinitionKey`.  Or if you're not ready to do Smart Gcal maybe just leave this undefined for now.

Reply:
I wasn't thinking about Smart Gcal yet. I think in GNIRS, as opposed to F2, the set of possible FPU parameters is fixed. There are 2 groups of FPU parameters, Slits and Other (for acquisition, or pupil), which if I understood they are mutually exclusive, thus the sum type.

From what you say I think I might need to rework it for Smart Gcal. I'll have a look at it now.

Reply:
Oh, I see now what you mean. I had the other Gnirs case class defined below in mind when making my previous comment. I'll comment this case class out.